features/events/daemon/src/main/java/org/opennms/netmgt/eventd/EventExpander.java

```diff
@@ -723,6 +726,22 @@ public synchronized void expandEvent(Event e) {
                 
                 e.setAlarmData(alarmData);
             }
+
+            if (econf.getParameterCollection() != null && econf.getParameterCount() > 0) {
+                if (e.getParmCollection() == null) {
+                    e.setParmCollection(new ArrayList<Parm>());
+                }
+                for (Parameter p : econf.getParameterCollection()) {
+                    Parm parm = new Parm();
+                    parm.setParmName(p.getName());
+                    Value v = new Value();
+                    v.setContent(p.getValue());
+                    v.setType("string");
+                    v.setEncoding("text");
+                    parm.setValue(v);
+                    e.addParm(parm);
```

Comment:
What's the expected behavior when an event already has a parameter with the same name?